projects/github-actions/repo-gardening/src/tasks/gather-support-references/index.js

```diff
@@ -1,63 +1,55 @@
 const debug = require( '../../debug' );
+const getComments = require( '../../get-comments' );
 
 /* global GitHub, WebhookPayloadIssue */
 
 /**
  * Search for a previous comment from this task in our issue.
  *
- * @param {GitHub} octokit - Initialized Octokit REST client.
- * @param {string} owner   - Repository owner.
- * @param {string} repo    - Repository name.
- * @param {string} number  - Issue number.
+ * @param {Array} issueComments - Array of all comments on that issue.
  * @returns {Promise<number>} Promise resolving to a number.
  */
-async function getListComment( octokit, owner, repo, number ) {
+async function getListComment( issueComments ) {
 	let commentID = 0;
 
 	debug( `gather-support-references: Looking for a previous comment from this task in our issue.` );
 
-	for await ( const response of octokit.paginate.iterator( octokit.rest.issues.listComments, {
-		owner,
-		repo,
-		issue_number: +number,
-	} ) ) {
-		response.data.map( comment => {
-			if (
-				comment.user.login === 'github-actions[bot]' &&
-				comment.body.includes( '**Support References**' )
-			) {
-				commentID = comment.id;
-			}
-		} );
-	}
+	issueComments.map( comment => {
+		if (
+			comment.user.login === 'github-actions[bot]' &&
+			comment.body.includes( '**Support References**' )
+		) {
+			commentID = comment.id;
+		}
+	} );
 
 	return commentID;
 }
 
 /**
  * Scan the contents of the issue as well as all its comments, get all support references, and add them to an array of references.
  *
- * @param {GitHub} octokit - Initialized Octokit REST client.
- * @param {string} owner   - Repository owner.
- * @param {string} repo    - Repository name.
- * @param {string} number  - Issue number.
+ * @param {GitHub} octokit      - Initialized Octokit REST client.
+ * @param {string} owner        - Repository owner.
+ * @param {string} repo         - Repository name.
+ * @param {string} number       - Issue number.
+ * @param {Array} issueComments - Array of all comments on that issue.
  * @returns {Promise<Array>} Promise resolving to an array.
  */
-async function getIssueReferences( octokit, owner, repo, number ) {
+async function getIssueReferences( octokit, owner, repo, number, issueComments ) {
 	const ticketReferences = [];
 	const supportIds = [];
 	const referencesRegexP = /[0-9]*-(?:chat|hc|zen|zd)/gim;
 
 	debug( `gather-support-references: Getting references from comments.` );
-	for await ( const response of octokit.paginate.iterator( octokit.rest.issues.listComments, {
-		owner: owner.login,
-		repo,
-		issue_number: +number,
-	} ) ) {
-		response.data.map( comment => {
+	issueComments.map( comment => {
+		if (
+			comment.user.login !== 'github-actions[bot]' &&
+			! comment.body.includes( '**Support References**' )
```

Comment:
This doesn't properly invert the test above.
```suggestion
			comment.user.login !== 'github-actions[bot]' ||
			! comment.body.includes( '**Support References**' )
```